tst/com/amazon/corretto/crypto/provider/test/AESGenerativeTest.java

```diff
@@ -281,7 +281,9 @@ private void updateAAD(Random r, ByteArrayOutputStream aadData) throws Throwable
       case 0:
         {
           jceCipherEncrypt.updateAAD(buf.duplicate());
-          amzCipherEncrypt.updateAAD(buf.duplicate());
+          final ByteBuffer amzBuf = buf.duplicate();
+          amzCipherEncrypt.updateAAD(amzBuf);
+          assertByteBufferEmpty(amzBuf);
```

Comment:
The change is for decryption. Shouldn't assertions be about buffers for decryption rather than encryption?

Reply:
I've updated *all* the buffer test cases that I could find. This bug crept in because our test coverage missed this case originally. So, my change not only introduces tests for this specific bug (failing to update position on decryption), it also improves the tests for all other cases to ensure we don't introduce a variant of this bug.

Reply:
It would be good to add it for aad update too: https://github.com/corretto/amazon-corretto-crypto-provider/blob/main/tst/com/amazon/corretto/crypto/provider/test/AESGenerativeTest.java#L524

Reply:
Thank you. I missed that one!